pages/dao/[symbol]/proposal/components/instructions/Saber/Deposit.tsx

```diff
@@ -0,0 +1,416 @@
+  serializeInstructionToBase64,
+  TOKEN_PROGRAM_ID,
+} from '@solana/spl-governance'
+import { ASSOCIATED_TOKEN_PROGRAM_ID, Token } from '@solana/spl-token'
+import { PublicKey, TransactionInstruction } from '@solana/web3.js'
+import Input from '@components/inputs/Input'
+import Select from '@components/inputs/Select'
+import useGovernanceAssets from '@hooks/useGovernanceAssets'
+import saberPoolsConfiguration, { Pool } from '@tools/sdk/saber/pools'
+import { getMintNaturalAmountFromDecimalAsBN } from '@tools/sdk/units'
+import { findAssociatedTokenAddress } from '@utils/associated'
+import { findATAAddrSync } from '@utils/ataTools'
+import { isFormValid } from '@utils/formValidation'
+import {
+  SaberPoolsDepositForm,
+  UiInstruction,
+} from '@utils/uiTypes/proposalCreationTypes'
+
+import useWalletStore from 'stores/useWalletStore'
+
+import { NewProposalContext } from '../../../new'
+import GovernedAccountSelect from '../../GovernedAccountSelect'
+
+async function deposit({
+  authority,
+  pool,
+  tokenAmountA,
+  tokenAmountB,
+  minimumPoolTokenAmount,
+}: {
+  authority: PublicKey
+  pool: Pool
+  tokenAmountA: BN
+  tokenAmountB: BN
+  minimumPoolTokenAmount: BN
+}): Promise<TransactionInstruction> {
+  const poolTokenMintATA = await findAssociatedTokenAddress(
+    authority,
+    pool.poolToken.mint
+  )
+
+  const [sourceA] = findATAAddrSync(authority, pool.tokenAccountA.tokenMint)
+  const [sourceB] = findATAAddrSync(authority, pool.tokenAccountB.tokenMint)
+
+  // TRICKS
+  // Have to add manually the toBuffer method as it's required by the @saberhq/stableswap-sdk package
+  // le = little endian
+  // 8 = 8 bytes = 64 bits
+  tokenAmountA.toBuffer = () => tokenAmountA.toArrayLike(Buffer, 'le', 8)
+  tokenAmountB.toBuffer = () => tokenAmountB.toArrayLike(Buffer, 'le', 8)
+  minimumPoolTokenAmount.toBuffer = () =>
+    minimumPoolTokenAmount.toArrayLike(Buffer, 'le', 8)
+
+  return depositInstruction({
+    config: {
+      authority: pool.swapAccountAuthority,
+      swapAccount: pool.swapAccount,
+      swapProgramID: saberPoolsConfiguration.saberStableSwapProgramId,
+      tokenProgramID: TOKEN_PROGRAM_ID,
+    },
+    userAuthority: authority,
+    sourceA,
+    sourceB,
+    tokenAccountA: pool.tokenAccountA.mint,
+    tokenAccountB: pool.tokenAccountB.mint,
+    poolTokenMint: pool.poolToken.mint,
+    poolTokenAccount: poolTokenMintATA,
+    tokenAmountA,
+    tokenAmountB,
+    minimumPoolTokenAmount,
+  })
+}
+
+const schema = yup.object().shape({
+  assetAccount: yup
+    .object()
+    .nullable()
+    .required('Governed account is required'),
+  uiTokenAmountA: yup.number().required('Amount for Token A is required'),
+  uiTokenAmountB: yup.number().required('Amount for Token B is required'),
+  uiMinimumPoolTokenAmount: yup
+    .number()
+    .moreThan(0, 'Minimum Pool Token Amount should be more than 0')
+    .required('Minimum Pool Token Amount is required'),
+})
+
+async function checkInitTokenAccount(
```

Comment:
I'm not a fan of functions pushing elements on provided array, but well, it has been done at others places in the realm project ^^'

Reply:
Yeah let's stay consistent with the Bullshit